src/style-spec/validate/validate_filter.js

```diff
@@ -13,7 +13,7 @@ export default function validateFilter(options) {
         return validateExpression(extend({}, options, {
             expressionContext: 'filter',
             // We default to a layerType of `fill` because that points to a non-dynamic filter definition within the style-spec.
-            valueSpec: options.styleSpec[`filter_${layerType || 'fill'}`]
```

Comment:
Good catch, much more readable!

I think that this change will have a conflict with the different approach @stepankuzmin is taking [here](https://github.com/mapbox/mapbox-gl-js/pull/11529/files#diff-270b52b46cd6599f7ffe53f4d49e9124f4a14c66d3e95b2f04cfd74dda1647bf).